test/e2e/specs/editor/various/keep-styles-on-block-transforms.spec.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Keep styles on block transforms', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Should keep colors during a transform', async ( {
+		page,
+		editor,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '## Heading' );
+
+		await page.click( 'role=button[name="Text"i]' );
```

Comment:
Seems like this test is flaky. Perhaps it's because sometimes the sidebar isn't opened. Try calling `editor.openDocumentSettingsSidebar()` before this statement?